src/main/java/appeng/client/render/items/ToolWirelessTerminalRender.java

```diff
@@ -0,0 +1,190 @@
+{
+
+    @Override
+    public boolean handleRenderType( ItemStack item, ItemRenderType type )
+    {
+        return true;
+    }
+
+    @Override
+    public boolean shouldUseRenderHelper( ItemRenderType type, ItemStack item, ItemRendererHelper helper )
+    {
+        return false;
+    }
+
+    @Override
+    public void renderItem( ItemRenderType type, ItemStack item, Object... data )
+    {
+        Entity itemLocation = item.getItemFrame();
+        if( itemLocation==null )
+        {
+            itemLocation = Minecraft.getMinecraft().thePlayer;
+        }
+        
+        boolean displayAntenna = ((ToolWirelessTerminal)item.getItem()).getIsUsable( item, itemLocation);
+        boolean hasPower = (( ToolWirelessTerminal) item.getItem()).hasPower( null,0.5,item );
+
+        IIcon border;
+        if( displayAntenna )
+        {
+            border = ExtraItemTextures.WirelessTerminal_Border.getIcon();
+        }
+        else
+        {
+            border = ExtraItemTextures.WirelessTerminal_Border_Inactive.getIcon();
+        }
+        IIcon scrollBar = ExtraItemTextures.WirelessTerminal_ScrollBar.getIcon();
+        IIcon screen = ExtraItemTextures.WirelessTerminal_Screen.getIcon();
+        IIcon icons = ExtraItemTextures.WirelessTerminal_Icons.getIcon();
+
+        AEColor color = ToolWirelessTerminal.getColor( item );
+        if( color==null )
+        {
+            screen = item.getIconIndex();
+        }
+
+        final Tessellator tessellator = Tessellator.instance;
+        GL11.glPushMatrix();
+        GL11.glPushAttrib( GL11.GL_ALL_ATTRIB_BITS );
+
+        //translate stuff for different item render types
+        if( type != ItemRenderType.INVENTORY )
+        {
+            if( type == ItemRenderType.EQUIPPED_FIRST_PERSON )
+            {
+                GL11.glTranslatef( 0.0F, 0.0F, 0.0F );
+            }
+            else if( type == ItemRenderType.EQUIPPED )
+            {
+                GL11.glTranslatef( 0.0F, 0.0F, 0.0F );
+            }
+            else
+            {
+                GL11.glTranslatef( -0.5F, -0.3F, 0.01F );
+            }
+        }
+        else
+        {
+            GL11.glColor4f( 1, 1, 1, 1.0F );
+            GL11.glScalef( 16F, 16F, 10F );
+            GL11.glTranslatef( 0.0F, 1.0F, 0.0F );
+            GL11.glRotatef( 180F, 1.0F, 0.0F, 0.0F );
+            GL11.glEnable( GL11.GL_ALPHA_TEST );
+        }
+
+        final float f12 = 0.0625F;
+
+        //Border, which is uncolored
+        subRenderItem( type, tessellator, border, f12);
+
+        if( hasPower )
+        {
+            RenderHelper.disableStandardItemLighting();
+        }
+
+        //If a terminal isn't colored, use the default icon which doesn't require icons or scrollbar
+        if( color!=null )
+        {
+            //Icons, which are dark colored
+            {
+                final int blackColor = color.blackVariant;
+                float r = ( blackColor >> 16 ) & 0xFF;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'r' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)